atd-etl/prefect/afd_ems_incidents/ems_incident_upload.py

```diff
@@ -0,0 +1,440 @@
+
+pp = pprint.PrettyPrinter(indent=4)
+
+kv_data = get_key_value(key="Vision Zero Development")
+environment_variables = json.loads(kv_data)
+
+# Retrieve the db configuration
+DB_USERNAME = None
+DB_PASSWORD = None
+DB_HOSTNAME = None
+DB_PORT = None
+DB_DATABASE = None
+AWS_ACCESS_KEY_ID = None
+AWS_SECRET_ACCESS_KEY = None
+AFD_S3_SOURCE_BUCKET = None
+AFD_S3_ARCHIVE_BUCKET = None
+AFD_S3_SOURCE_PREFIX = None
+AFD_S3_ARCHIVE_PREFIX = None
+EMS_S3_SOURCE_BUCKET = None
+EMS_S3_ARCHIVE_BUCKET = None
+EMS_S3_SOURCE_PREFIX = None
+EMS_S3_ARCHIVE_PREFIX = None
+
+if False:
+    # Retrieve the db configuration
+    DB_USERNAME = environment_variables["AFD_DB_USERNAME"]
+    DB_PASSWORD = environment_variables["AFD_DB_PASSWORD"]
+    DB_HOSTNAME = environment_variables["AFD_DB_HOSTNAME"]
+    DB_PORT = environment_variables["AFD_DB_PORT"]
+    DB_DATABASE = environment_variables["AFD_DB_DATABASE"]
+    AWS_ACCESS_KEY_ID = environment_variables["AWS_ACCESS_KEY_ID"]
+    AWS_SECRET_ACCESS_KEY = environment_variables["AWS_SECRET_ACCESS_KEY"]
+    AFD_S3_SOURCE_BUCKET = environment_variables["AFD_S3_SOURCE_BUCKET"]
+    AFD_S3_ARCHIVE_BUCKET = environment_variables["AFD_S3_ARCHIVE_BUCKET"]
+    AFD_S3_SOURCE_PREFIX = environment_variables["AFD_S3_SOURCE_PREFIX"]
+    AFD_S3_ARCHIVE_PREFIX = environment_variables["AFD_S3_ARCHIVE_PREFIX"]
+    EMS_S3_SOURCE_BUCKET = environment_variables["EMS_S3_SOURCE_BUCKET"]
+    EMS_S3_ARCHIVE_BUCKET = environment_variables["EMS_S3_ARCHIVE_BUCKET"]
+    EMS_S3_SOURCE_PREFIX = environment_variables["EMS_S3_SOURCE_PREFIX"]
+    EMS_S3_ARCHIVE_PREFIX = environment_variables["EMS_S3_ARCHIVE_PREFIX"]
+else:
+    DB_USERNAME = os.getenv("AFD_DB_USERNAME")
+    DB_PASSWORD = os.getenv("AFD_DB_PASSWORD")
+    DB_HOSTNAME = os.getenv("AFD_DB_HOSTNAME")
+    DB_PORT = os.getenv("AFD_DB_PORT")
+    DB_DATABASE = os.getenv("AFD_DB_DATABASE")
+    AWS_ACCESS_KEY_ID = os.getenv("AWS_ACCESS_KEY_ID")
+    AWS_SECRET_ACCESS_KEY = os.getenv("AWS_SECRET_ACCESS_KEY")
+    AFD_S3_SOURCE_BUCKET = os.getenv("AFD_S3_SOURCE_BUCKET")
+    AFD_S3_ARCHIVE_BUCKET = os.getenv("AFD_S3_ARCHIVE_BUCKET")
+    AFD_S3_SOURCE_PREFIX = os.getenv("AFD_S3_SOURCE_PREFIX")
+    AFD_S3_ARCHIVE_PREFIX = os.getenv("AFD_S3_ARCHIVE_PREFIX")
+    EMS_S3_SOURCE_BUCKET = os.getenv("EMS_S3_SOURCE_BUCKET")
+    EMS_S3_ARCHIVE_BUCKET = os.getenv("EMS_S3_ARCHIVE_BUCKET")
+    EMS_S3_SOURCE_PREFIX = os.getenv("EMS_S3_SOURCE_PREFIX")
+    EMS_S3_ARCHIVE_PREFIX = os.getenv("EMS_S3_ARCHIVE_PREFIX")
+
+
+
+@task
+def get_timestamp():
+    current = datetime.now()
+    return time.mktime(current.timetuple())
+
+
+@task
+def get_most_recent_email():
+    """
+    Find the most recently updated file in the bucket. This will be the newest email.
+    :return: string
+    """
+
+    session = boto3.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
+    s3 = session.resource("s3")
```

Comment:
These `boto3` session make it thread safe which is needed if we want to do dask-enabled parallelism down the road.